### PR TITLE
fix: fix engine replica deletion and replica varification issues

### PR DESCRIPTION
**Pull request by shuo-wu**

### CODE REVIEW

File: pkg/spdk/engine.go [hunk 1]
```diff
@@ -1234,6 +1234,19 @@ func (e *Engine) ReplicaDelete(spdkClient *spdkclient.Client, replicaName, repli
...
+		if err = e.initiator.Suspend(true, true); err != nil {
+			return errors.Wrapf(err, "failed to suspend NVMe initiator before removing replica %s base bdev from engine %s RAID", replicaName, e.Name)
+		}
+		defer func() {
+			if err = e.initiator.Resume(); err != nil {
+				err = errors.Wrapf(err, "failed to suspend NVMe initiator after removing replica %s base bdev from engine %s RAID", replicaName, e.Name)
+			}
+		}()
+	}
+
```
Comment (derekbit) [inline pkg/spdk/engine.go:1252]: What do we need to suspend frontend for the replica deletion?

**Comment by derekbit** [comment]
@shuo-wu Can the PR be reviewed?

File: pkg/spdk/engine.go (hunk 1 above)
Comment (shuo-wu) [reply pkg/spdk/engine.go:1252]: Let's remove this commit. 
When Longhorn deletes a replica from the engine, spdk_tgt will print out lots of errors like `nvme_tcp.c:2177:nvme_tcp_qpair_process_completions: *ERROR*: Failed to flush tqpair=0x1826e70 (9): Bad file descriptor`. I was thinking adding a frontend suspend would avoid it. but after testing it today, the answer is NO.

**Review by derekbit — APPROVED** [review]

**Review by derekbit — APPROVED** [review]
LGTM

**Review by derekbit — APPROVED** [review]

**Pull request merged**